### PR TITLE
Match multiple file extensions 

### LINKED ISSUE #50: Matching multiple file extensions automatically
Quick sketch for this:
```javascript
const fileName = 'phpcs.xml.dist.sample';
const fileExtensions = [];
for (let index = 0; index < fileName.length; index += 1) {
  if (fileName[index] === '.') fileExtensions.push(fileName.slice(index + 1));
}
```

### PR DESCRIPTION
**Pull request by zm-cttae**
Closes #50. Prior art:

https://github.com/microsoft/vscode/blob/5235c6b/src/vs/editor/common/services/getIconClasses.ts#L57-L58

### CODE REVIEW

File: src/lib/replace-icon.js [hunk 1]
```diff
@@ -31,22 +31,30 @@ function replaceIcon(iconEl, fileName, itemRow, provider, iconPack) {
...
-  // Get file extension.
-  const fileExtension = fileName.match(/.*?[.](?<ext>xml.dist|xml.dist.sample|yml.dist|\w+)$/)?.[1];
+  // Get file extensions.
+  const fileExtensions = [];
+  // Avoid doing an explosive combination of extensions for very long filenames
+  // (most file systems do not allow files > 255 length) with lots of `.` characters
+  // https://github.com/microsoft/vscode/issues/116199
+  if (fileName.length <= 255) {
+    for (let i = 0; i < fileName.length; i += 1) {
+      if (fileName[i] === '.') fileExtensions.push(fileName.slice(i + 1));
```
Comment (Claudiohbsantos) [inline src/lib/replace-icon.js:41]: To confirm the intended behavior, for a file `name.ext1.ext2.ext3` we expect `fileExtensions` to be `['ext1.ext2.ext', 'ext2.ext3', 'ext.3'] ` ?

File: src/lib/replace-icon.js [hunk 2]
```diff
@@ -94,18 +102,22 @@ function lookForMatch(fileName, lowerFileName, fileExtension, isDir, isSubmodule
     return iconMap.folderNames[lowerFileName];
 
   // Look for extension in fileExtensions and languageIds.
-  if (iconMap.fileExtensions[fileExtension] && !isDir && !isSubmodule)
-    return iconMap.fileExtensions[fileExtension];
-  if (iconMap.languageIds[fileExtension] && !isDir && !isSubmodule)
-    return iconMap.languageIds[fileExtension];
+  for (const ext of fileExtensions) {
```
Comment (Claudiohbsantos) [inline src/lib/replace-icon.js:105]: Just confirming that the intention here is to match the first (left-most) extension that matches a known icon. Is that the idea?

**Review by Claudiohbsantos — COMMENTED** [review]
Hey @zm-cttae ! Thanks for the pull request, and I'm very sorry it has sat open for so long. 

I took a look at the ticket and I'm not entirely clear on what problem we're trying to solve here. Could you give me a couple of examples of files that are being missed by the current logic but that would be correctly picked up with these changes?

Thanks!

File: src/lib/replace-icon.js (hunk 1 above)
Comment (zm-cttae) [reply src/lib/replace-icon.js:41]: Sounds right. This is the same implementation that's present in core VS Code UI

**Comment by zm-cttae** [comment]
Before this PR, any `fileExtensions` entry with one or more dot was broken. This fixes all of them. You can test this on a repository with `spec.ts` `test.ts` `spec.js` `test.js` extensions

File: src/lib/replace-icon.js (hunk 2 above)
Comment (zm-cttae) [reply src/lib/replace-icon.js:105]: Correct

**Pull request merged**